src/command/archive/push/file.c

```diff
@@ -117,21 +151,69 @@ archivePushFile(
                 compressible = false;
             }
 
-            // If there is a cipher then add the encrypt filter
-            if (cipherType != cipherTypeNone)
+            // Initialize per-repo destination files
+            StorageWrite **destination = memNew(sizeof(StorageWrite *) * repoTotal);
+
+            for (unsigned int repoIdx = 0; repoIdx < repoTotal; repoIdx++)
             {
-                ioFilterGroupAdd(
-                    ioReadFilterGroup(storageReadIo(source)),
-                    cipherBlockNew(cipherModeEncrypt, cipherType, BUFSTR(cipherPass), NULL));
-                compressible = false;
+                // Does this repo need a copy?
+                if (destinationCopy[repoIdx])
+                {
+                    // Create destination file
+                    destination[repoIdx] = storageNewWriteP(
+                        storageRepoIdxWrite(repoIdx),
+                        strNewFmt(STORAGE_REPO_ARCHIVE "/%s/%s", strZ(repoData[repoIdx].archiveId), strZ(archiveDestination)),
+                        .compressible = compressible && repoData[repoIdx].cipherType == cipherTypeNone);
+
+                    // If there is a cipher then add the encrypt filter
+                    if (repoData[repoIdx].cipherType != cipherTypeNone)
+                    {
+                        ioFilterGroupAdd(
+                            ioWriteFilterGroup(storageWriteIo(destination[repoIdx])),
+                            cipherBlockNew(
+                                cipherModeEncrypt, repoData[repoIdx].cipherType, BUFSTR(repoData[repoIdx].cipherPass), NULL));
```

Comment:
It matters where it is set. In this case only the write object can be remote, so setting compressible on the source is useless (but harmless). I can't quite figure out the rest of what you wrote here.

What I think you are saying is: since the encryption filter is being pushed to the remote on the write side compressible really only depends on whether the source is compressed or not. So, we can either remove the compressible dependency on encryption, or not push the encryption filter to the remote (larger code change).